File: ios/Classes/ThreatCenterApiImpl.swift
import Flutter
import SecurityToolkit

public class ThreatCenterApiImpl: NSObject, FlutterPlugin, ThreatCenterApi {
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let messenger : FlutterBinaryMessenger = registrar.messenger()
        let api : ThreatCenterApi & NSObjectProtocol = ThreatCenterApiImpl.init()
        ThreatCenterApiSetup.setUp(binaryMessenger: messenger, api: api)
    }

    // MARK: - Threat Center Api
    
    func areRootPrivilegesDetected() throws -> Bool {
        ThreatDetectionCenter.areRootPrivilegesDetected
    }
    
    func areHooksDetected() throws -> Bool {
        ThreatDetectionCenter.areHooksDetected
    }
    
    func isSimulatorDetected() throws -> Bool {
        ThreatDetectionCenter.isSimulatorDetected
    }
}

File: android/src/main/kotlin/com/exxeta/security_toolkit/ThreatCenterApiImpl.kt

package com.exxeta.security_toolkit

import ThreatCenterApi
import com.exxeta.securitytoolkit.ThreatDetectionCenter
import io.flutter.embedding.engine.plugins.FlutterPlugin

class ThreatCenterApiImpl : FlutterPlugin, ThreatCenterApi {
  private lateinit var threatDetectionCenter: ThreatDetectionCenter

  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    threatDetectionCenter = ThreatDetectionCenter(binding.applicationContext)
    ThreatCenterApi.setUp(binding.binaryMessenger, this)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    ThreatCenterApi.setUp(binding.binaryMessenger, null)
  }

  // MARK: - ThreatCenterApi

  override fun areRootPrivilegesDetected(): Boolean {
    return threatDetectionCenter.areRootPrivilegesDetected
  }

  override fun areHooksDetected(): Boolean {
    return threatDetectionCenter.areHooksDetected
  }

  override fun isSimulatorDetected(): Boolean {
    return threatDetectionCenter.isSimulatorDetected
  }
}